Doppler.HtmlEditorApi/Repositories.DopplerDb/DopplerCampaignContentRepository.cs

```diff
@@ -59,23 +60,32 @@ public async Task<CampaignState> GetCampaignState(string accountName, int campai
             IdCampaign: campaignId
         ));
 
-        if (campaignStateData == null || !campaignStateData.OwnCampaignExists)
-        {
-            return new NoExistCampaignState();
-        }
-
         // For information about Doppler's status code, check out here
         // https://github.com/MakingSense/Doppler/blob/develop/Doppler.Transversal/Classes/CampaignStatusEnum.cs
         var campaignStatus = campaignStateData.Status is DopplerCampaignStatusDraft or
             DopplerCampaignStatusABDraft ? CampaignStatus.Draft
             : campaignStateData.Status == DopplerCampaignStatusInWinnerInABSelectionProcess ? CampaignStatus.InWinnerInABSelectionProcess
             : CampaignStatus.Other;
 
+        if (campaignStateData == null || !campaignStateData.OwnCampaignExists)
+        {
+            return new NoExistCampaignState();
+        }
+
+        if (campaignStateData.TestType == 0)
+        {
+            return new NoTestABCampaignState(campaignStateData.ContentExists, campaignStateData.EditorType, campaignStatus);
+        }
+
+        var campaignTestABCondition = campaignStateData.TestType is DopplerCampaignTestTypeSubject ? TestABCondition.TypeTestABSubject
+        : TestABCondition.TypeTestABContent;
+
         return new CampaignState(
```

Comment:
Related to: https://github.com/FromDoppler/doppler-html-editor-api/pull/113/files#r872343322

```suggestion
        return new TestABCampaignState(
```